### ISSUE TITLE
Vymyslet tričko pro "Python v ČR" komunitu

### ISSUE THREAD

**Issue by honzajavorek** [pyvec/zapojse#29]
Pracujeme na tom s https://www.jack.cz/ Jejich návrh je “I SPEAK PARSELTONGUE”

![13663580_10209236276439190_1913680535_o](https://cloud.githubusercontent.com/assets/283441/17832907/875acbb2-670f-11e6-9ac2-253713ba63b6.jpg)

Já jsem si říkal, že to je dobré, ale nemusí to oslovit každého. Takže je otázka, zda vymyslet něco jiného, nebo udělat tohle plus jedno "nudné" tričko, které osloví úplně každého. Nápady, názory?

**Comment by hroncok** [comment]
B a D na druhém řádku nejde přečíst

**Comment by hroncok** [comment]
Jinak „nudné“ mi přijde i tohle (parseltongue super, ale designově nuda)

**Comment by radekholy24** [comment]
Co nějaké narážky na Zen Buddhismus a Zen of Python?

Jinak mně se asi nejvíc líbí ten design na C a D... Barevně je to hezký všechno...

**Comment by brabitom** [comment]
- "I SPEAK PARSELTONGUE" je opravdu dobrý nápad
- bylo by možné by C (D) v official. žlutomodré barvě?

**Comment by zuzejk** [comment]
jen pro inspiraci https://www.teepublic.com/show/374625-python-programmer-heartbeat-t-shirt-and-hoodie

**Comment by hanpari** [comment]
Pokud to je pro českou komunitu, nebylo by jaksi vhodnější použít češtinu? Však víte, rodný jazyk :)

**Comment by hroncok** [comment]
Klidně anglicky, ale ať jde poznat, že jde o českou komunitu.

**Comment by radekholy24** [comment]
Zepředu česky, zezadu anglicky :)

**Comment by OndrejIT** [comment]
za me C D v ofiko barvach...

**Comment by radekholy24** [comment]
"Curly braces? No, thanks!" :-)

**Comment by milonline-eu** [comment]
Starsi nevedej, co je PARSELTONGUE...

**Comment by jakolar** [comment]
Tricko oslovuje predevsim cilovku Potter & Python a zobecneni, pohledem zpetne, ma smysl. Co stylizace loga napr. http://orig00.deviantart.net/95f9/f/2014/354/f/8/ehs_cs_118_python_logo_by_epula_daug-d8al651.png. Silne byvaji sdeleni, ktere nejsou primocara, napr. zminene Curly braces - No thanks, Maybe Later...

**Comment by radekholy24** [comment]
Třeba český, dvouocasý lev se žlutým a modrým hadem místo ocasu by byl asi moc, co? :)

**Comment by honzajavorek** [comment]
Lva už jsem kreslil na záda trika na PyCon 2015. Blbý je, že zahraniční
toho lva neznají, tak ten vtip není úplně zřejmý a vypadá to, že mu rostou
ze zadku dva hadi a v tom pak člověk hledá symboliku...

**Comment by aleszoulek** [comment]
No, když jsem zjistil, na co je to narážka, tak  se mi návrh líbí. Ale neznal jsem...

Jinými slovy, radši zajímavý motiv, který pochopí míň lidí, než obecný konferenční logo, kterých ma člověk ve skříni tucty.

**Comment by honzajavorek** [comment]
Mimochodem nějaký návrhy se objevují i [tady pod příspěvkem na FB](https://www.facebook.com/groups/pyonieri/permalink/1205034902842018/), tak aby to nezapadlo.

**Comment by radekholy24** [comment]
Nemohlo by to tričko bejt modrožlutý? Ty bílý a černý "reklamní" trička jsou opruz :-)

**Comment by jakolar** [comment]
Ok, co takhle si pohrat s podobnosti bites - bytes?
- "Beware of Python bytes"
- "Python bytes running in my veins"

**Comment by brabitom** [comment]
> @PyDeq Nemohlo by to tričko bejt modrožlutý? Ty bílý a černý "reklamní" trička jsou opruz :-)

Pravda, to by mohlo vypadat dobře. Třeba logo Pythonu, v origo barvách, pootočené a roztáhnuté přes celé triko, které by tak bylo modro žluté.

**Comment by flaiming** [comment]
Tady jsou taky pěkné nápady :) https://www.reddit.com/r/Python/comments/1pw3av/what_cool_python_code_could_you_fit_on_a_tshirt/

Mé nejoblíbenější:

`import this`

`import antigravity`

http://i.imgur.com/42aY919.jpg

`four spaces`

http://www.cafepress.co.uk/mf/90622228/hip-hip-hooray-programming-array_tshirt?productId=1696200997

**Comment by radekholy24** [comment]
Dneska mě zaujalo tohle tričko... Co se inspirovat a udělat něco, jako že nám místo srdce tluče Python?... Ale to už tu taky bylo, předpokládam...
![](https://ibin.co/2soe1cKUnCzv.png)

**Comment by hroncok** [comment]
A ten worldcloud je blbý nápad? Tady u toho písma mám probléms českými znaky, ale v případě, že by se to líbilo, šlo by to vyřešit. Je tam @encukou i @honzajavorek. Texty z pyvo.cz/město.

![pyvo-triko](https://cloud.githubusercontent.com/assets/2401856/17983686/2727d016-6b0e-11e6-90e4-bc8fb3a17c96.png)

**Comment by radekholy24** [comment]
To se mi líbí zatím ze všeho nejvíc :-)

**Comment by hroncok** [comment]
Tady s jiným fontem, který umí háčkyčárky a s přidáním nějakých stopwords (na, se).

![pyvo-triko-font](https://cloud.githubusercontent.com/assets/2401856/17983797/aed9d70c-6b0e-11e6-8485-64700a3ff86a.png)

**Comment by honzajavorek** [comment]
Javorek Republic, to zní skvěle ;)

**Comment by flaiming** [comment]
Super! :) Btw kde se tam vzalo to "Ostrovského"?

**Comment by encukou** [comment]
Wordcloud bych nosil! Ale nerad bych na něm byl takhle velkým písmem. Data by to chtělo trochu poladit – např Na Věnečku jako jedno slovo, vyhodit 38a, a tak.
Nechceme tam dát i slova z pyladies/djangogirls materiálů?
@flaiming, Ostrovského 38a je z adresy Pražského Pyva.

**Comment by jakolar** [comment]
Idea s Wordcloudem se mi libi, kazde tricko muze byt jine.

**Comment by hroncok** [comment]
Tady je **hrozně naprasený** kód k tomu a použité logo:
<details>

``` python
import matplotlib.pyplot as plt
from wordcloud import WordCloud, STOPWORDS, ImageColorGenerator
import numpy as np
from PIL import Image
from scipy.misc import imread

alice_coloring = np.array(Image.open("logo.png"))

more_stopwords = {'na', 'se'}
STOPWORDS = STOPWORDS.union(more_stopwords)


with open('packages.txt', 'r') as f:
    packages = f.read()

wordcloud = WordCloud(
                      #font_path='flux.ttf',
                      stopwords=STOPWORDS,
                      background_color='black',
                      width=18000,
                      height=14000,
                      mask=alice_coloring
                     ).generate(packages)

image_colors = ImageColorGenerator(alice_coloring)


# show
plt.imshow(wordcloud)
plt.axis("off")
plt.figure()
# recolor wordcloud and show
# we could also give color_func=image_colors directly in the constructor
plt.imshow(wordcloud.recolor(color_func=image_colors))
plt.axis("off")
plt.figure()
plt.imshow(alice_coloring, cmap=plt.cm.gray)
plt.axis("off")
plt.show()
```

![logo](https://cloud.githubusercontent.com/assets/2401856/18002959/53f87e92-6b8b-11e6-8346-2129ff64fb53.png)
</details>

**Comment by encukou** [comment]
Ten kód by to chtělo vyčistit a dát na záda :)

**Comment by flaiming** [comment]
Tyjo, to jsem nevěděl, že na to existuje balík :) To musím použít i jinde.

**Comment by zuzejk** [comment]
vypadá to velice fajne! ale opravdu by to chtělo doplnit některá klíčová slova, tak jak píše encukou. Asi ani nemusim psát které...

**Comment by zuzejk** [comment]
ještě mě napadlo udělat obrázek dopravní značky "pozor zvěř" a do ní dát místo jelena logo pythonu :)

**Comment by encukou** [comment]
A co takhle vpašovat do wordcloudu nějaký slogan?

![wordcloud-pyvo-bude-fs8](https://cloud.githubusercontent.com/assets/302922/18097897/31eef21a-6ee0-11e6-98ef-8491dbfb7d4b.png)

**Comment by aleszoulek** [comment]
Libi moc!

A "pyvo bude" je taky fajn. I napad s kodem na zada. :-)

AZ

Dne út 30. 8. 2016 18:33 uživatel Petr Viktorin notifications@github.com
napsal:

**Comment by frenzymadness** [comment]
Poslení obrázek od @encukou vypadá skvěle a nápad s kódem na záda je taky bomba! :+1:

**Comment by honzajavorek** [comment]
To pozor zver si snad udelam i klidne jen sam pro sebe! To je super.

**Comment by jakolar** [comment]
Pokud je wordcloud finálně rozhodnut, zvažte zda-li ho budete chtít personalizovat. V praxi by to znamenalo obdržet od vás x-PNG soborů a napárovanou tabulku velikostí.

**Comment by hroncok** [comment]
To se neděje ve vektorech?

**Comment by jakolar** [comment]
Pokud jde o digitální tisk, tak ne, neděje.

**Comment by hroncok** [comment]
Jaké je ideální rozlišení?

**Comment by jakolar** [comment]
8400 × 9600

**Comment by zuzejk** [comment]
nějaký progres? má to být do pyconu?

**Comment by jakolar** [comment]
Také jsem na to myslel. Za vás je důležité se rozhodnout, pokud budete chtít unikátní trička nebo jednotný motiv.

**Comment by honzajavorek** [comment]
@hroncok Nechceš si to tak nějak celé vzít na starost? Podle mě je nápadů milion, ale chybí osoba, která by to nějak posbírala, zvážila, rozhodla a s @jankolario pak dotáhla. Finance z Pyvce zařídíme přes @martinbilek, ale chce to mít odhad.

**Comment by hroncok** [comment]
Do PyConu? To určitě nestíhám.

**Comment by hroncok** [comment]
Ten kód kterým jsme to generoval nedokážu přimět k vygenerování vektoru ani obří bitmapy. Pokud bude obrázek jen jeden, tzn ten poslední on @encukou, můžu to ručně předělat na vektor. Pokud bychom ale chtěli generovat různé, takhle to nepůjde.

Zároveň pokud budeme chtít jen jednu variantu, můžu poskytnout takhle hodně barevnou a pak dvoubarevnou pro sítotisk, který by vyšel levněji, ale je možné, že to nebude tak krásné.

Pokyny?

**Comment by jakolar** [comment]
@hroncok jaký je limit rozlišení, u které to přestává korektně fungovat?

**Comment by hroncok** [comment]
Umím si to jen zobrazit na obrazovku. Moje přílohy jsou de facto screenshoty v největším rozlišení, co jsem z toho vytřískal.

**Comment by zuzejk** [comment]
no, ačkoliv chápu, jak je dělaný kód, tak přiznám se, že mě by teda hodně mrzelo že tam není ani jeden malý prd týkající se holek v python komunitě.

**Comment by hroncok** [comment]
Není problém do toho textu narvat nová slova. Django Girls, PyLadies, holky, ženy, dámy, slečny, paní, cokoliv.

**Comment by encukou** [comment]
Generovat obří bitmapu umím (`WordCloud(scale=...)`). Mám rozdělaný kód, který bere slova z:
- pyvo.cz
- blog.python.cz
- pyladies materiálů

s tím že by pak měl ty slova podle "původu" dát do žluté nebo modré části. Do Pyva to snad zvládnu dokódit, na Pyvu můžeme doladit detaily (nař. co tam bude za slova). Mám na tom zamakat?

Chceme trička individuální? Mně přijde že to ani nemá moc smysl, a byla by to celkem otrava vytvářet. Na "Pyvo bude" jsem ten skript spouštěl dokud to nebylo ve správném pořadí; pro padesát lidí se mi to dělat nechce.

[Tady](http://imgur.com/a/DQWnn) je obrázek se slovama i z blogu  a materiálů. Originál má 10570x10514 px.

**Comment by honzajavorek** [comment]
BTW na FreeWear.org mají tohle. Nedávám to úplně jako návrh, spíš mě to (asi jako jediný co tam mají) pobavilo:

![fw0187](https://cloud.githubusercontent.com/assets/283441/18614412/6d508690-7d8e-11e6-83bb-2721ad2cff31.png)

**Comment by encukou** [comment]
Další verze:  ![wordcloud](https://cloud.githubusercontent.com/assets/302922/18724865/41730522-803e-11e6-870f-16a0d1db5d28.png)

Co vylepšit:
- [x] "Petr Viktorin" je tam příliš výrazný

**Comment by hroncok** [comment]
Nepřipadáš si tam příliš výrazný?

**Comment by honzajavorek** [comment]
Co jsou všechno momentálně zdroje slov? Jak těžké je přidat nový zdroj?

**Comment by encukou** [comment]
Aktuálně je to blog (modrá část loga), pyvo.cz (žlutá) a materiály pro PyLadies (přispívají do obou). Nový zdroj není těžké přidat. Jen v podobě jak to mám teď to funguje na mém počítači, snad brzo najdu čas to napravit aby mohli ostatní experimentovat.

**Comment by jakolar** [comment]
Jakmile budete mít hotový tiskový soubor, dejte vědět. Tisk přes digitál trvá cca. 7 dní, u síta je to cca. 12 dní.

**Comment by encukou** [comment]
Kód je tady v repozitáři (i když není moc hezký): https://github.com/pyvec/py-shirt-wordcloud

@honzajavorek, jestli chceš přidat nový zdroj, řekni prosím jaký. Jdeme do finiše.

**Comment by hroncok** [comment]
Umím z toho inkscapem udělat křivky. Zkusím to udělat přes potrace, aby se
to nemuselo dělat ručně.

**Comment by hroncok** [comment]
Je to zautomatizované. Kód v https://github.com/pyvec/py-shirt-wordcloud a výsledek v https://github.com/pyvec/py-shirt-wordcloud/tree/output

**Comment by encukou** [comment]
Pošlu mail na PSF-trademarks@python.org, jestli je to v pohodě :)
Na brněnském Pyvu (ve čtvrtek) bych poprosil dav o poslední připomínky, takže finální verze by byla v pátek.

**Comment by honzajavorek** [comment]
@encukou Přemýšlel jsem, jestli tam nedat i python.cz, ale asi by to moc nezměnilo a přibyla by tam spousta názvů firem. Zajímavější by bylo vytáhnout obsah Slack messages, GitHub comments nebo něco, ale to je na celý večer a ten já nemám.

Taky je možná otázka, jestli tam chceme vůbec nějaké názvy firem (aby to bylo spravedlivé - asi by tam měly být jen pokud tričko sponzorují), takže bych případné vyházel.

**Comment by encukou** [comment]
Jo, na python.cz jsem se taky koukal, ale holt tam nejsou dlouhé texty. Ze Slacku možná vyleze nějaká  :tlacenka:, vyžádal jsem si export dat.

Sponzoruje to tričko někdo?

**Comment by encukou** [comment]
Když přidám data ze Slacku, vypadá to takhle: ![wordcloud-small](https://cloud.githubusercontent.com/assets/302922/18927697/42bb7548-85bd-11e6-9eb1-dbfc5c8cba81.png)

Přijde mi že to ani nemá cenu zkoušet čistit.

**Comment by encukou** [comment]
... ale kdyby to někdo chtěl zkusit, je to ve [větvi slack](https://github.com/pyvec/py-shirt-wordcloud/tree/slack).

**Comment by honzajavorek** [comment]
> Sponzoruje to tričko někdo? 

Pokud vím, tak je to jen mezi Pyvcem a jack.cz. Pyvec logo bych tam ale nedával. Teď už nevím, jaká byla domluva s jackem - @jankolario ? Předpokládám, že Pyvec to nějak nakoupí a trička budou lidem k prodeji a zaplatí se svou cenou. Nebo to vlastně jack umí úplně na objednávku a každému to posílá zvlášť?

**Comment by hroncok** [comment]
Ještě mě napadlo:

![wordcloud_black](https://cloud.githubusercontent.com/assets/2401856/18951084/70cf54a0-8643-11e6-8358-9da993e6e47c.png)
![wordcloud_blue](https://cloud.githubusercontent.com/assets/2401856/18951085/70d51fac-8643-11e6-9fdd-4d34f088ab71.png)
![wordcloud_green](https://cloud.githubusercontent.com/assets/2401856/18951086/70de888a-8643-11e6-8e1a-45f33b5046ed.png)
![wordcloud_red](https://cloud.githubusercontent.com/assets/2401856/18951087/712e9bf4-8643-11e6-859e-9704c123e13f.png)
![wordcloud_white](https://cloud.githubusercontent.com/assets/2401856/18951090/7138f770-8643-11e6-9c5e-51fb914fe775.png)

**Comment by frenzymadness** [comment]
@hroncok černá je asi nej, ale jako motorkář a metalista asi nemám v tomhle plné hlasovací právo :D

**Comment by encukou** [comment]
Feedback z PSF:

> While the wordcloud looks nice, it does change the outline of
> the logo. You could fix this by either adding a background
> color in the shape of the full Python logo shape or by introducing
> a thin line with the shape.

**Comment by frenzymadness** [comment]
Nápad na změnu, která by mohla vyhovovat PSF: Co takhle udělat wordcloud inverzně - tedy pozadí oněch tvarů udělat v barvě (modrá/žlutá) a text na něm černou barvou? Tím bychom docílili toho, že logo bude mít přesný tvar, jak vyžaduje PSF. Otázkou je, jak moc to ovlivní čitelnost textu, ale za zkoušku to stojí.

**Comment by hroncok** [comment]
Bez okraje:

![wordcloud_neg](https://cloud.githubusercontent.com/assets/2401856/22603446/bd0f5dde-ea47-11e6-8b20-1243cf82e23f.png)


Další možnost je vygenerovat ten wordcloud z nějakýho trochu smrsklýho tvaru, aby ta písmenka nebyla nalepaná až k okraji.

**Comment by benabraham** [comment]
@hroncok To bude na sítotisku nic moc obávám se. Moc detailů. Nebudou vidět. Navíc je tam pořád dost potisklé plochy a vznikne efekt „mám na hrudi kus igelitu“.

**Comment by benabraham** [comment]
Ten princip je ale dobrej, jen to dotáhnout (a zbavit se slov jako „takže“)

**Comment by hroncok** [comment]
Jen si tady odložím:

![dsc_0782](https://cloud.githubusercontent.com/assets/2401856/23826493/3c70b426-069d-11e7-8d1d-881a9af2866b.JPG)

Turris Omnia má tričko s hodně vysokým detailem a u toho digitálního tisku není efekt igelitové placky na hrudi tak hrozný.

**Comment by benabraham** [comment]
@hroncok To je tedy digitál a ne sítotisk? Mě přijde, že do dokonalosti tomu poslednímu návrhu chybí dotáhnout ty obsah (slova) + jejich velikosti (četnost != důležitost pro tričko) a zvolit font tak, aby tam bylo co nejmíň tý potištěný plochy. 

Zároveň by to chtělo začít ukazovat i na tričku (tedy velikost a umístění), fajn [nástroj:](https://placeit.net/c/apparel/stages/heather-t-shirt-mockup-of-a-young-man-walking-around-a8662)

**Comment by honzajavorek** [comment]
@encukou @hroncok @benabraham nechcete tohle teď na sprintu oživit?

**Comment by hroncok** [comment]
S okrajem (obsah slov neřešit, to English se musí potlačit).

![wordcloud_reversed_faked svg](https://user-images.githubusercontent.com/2401856/29741073-0637c78a-8a66-11e7-8f47-6aea8aab2168.png)

**Comment by hroncok** [comment]
![wordcloud_ svg](https://user-images.githubusercontent.com/2401856/29743472-4bc33e1a-8a93-11e7-971c-f61a04514617.png)


![triko](https://user-images.githubusercontent.com/2401856/29743481-705ee0f8-8a93-11e7-8e7e-1b94384111c6.png)

**Comment by hroncok** [comment]
![triko2](https://user-images.githubusercontent.com/2401856/29743509-f7a1f53c-8a93-11e7-8e52-57221bbed423.png)

**Comment by hroncok** [comment]
@jakolar řešíme to stále s Tebou? jack.cz je down a nevím jestli dočasně nebo na pořád :(

**Comment by SteanlyCZ** [comment]
Jak to dopadlo? Dá se někde sehnat?

**Comment by benabraham** [comment]
Je to stále otevřené.

**Comment by hroncok** [comment]
Prišlo mi, že většině lidí, se kterými jsem mluvil osobně, přijde současný
grafický návrh nemoderní a nudný, a ztratil jsem chuť se tomu věnovat.

Dne 9. 2. 2018 8:34 dop. napsal uživatel "Daniel Srb" <
notifications@github.com>:

**Comment by jakolar** [comment]
@hroncok Jack v tuto chvíli funguje, nicméně jeho provoz je nezávislý na tom, že Vám pomůžeme s tiskem u OP Tiger. Pokud je to za Vás v takovém limbu, tak navrhuji vzít TOP5 inspirací z https://www.redbubble.com/shop/programmer+t-shirts zahlasovat o nich a vítězný návrh realizovat.

**Comment by benabraham** [comment]
@hroncok 27. nejprodávanější tričko v programátorskejch :) https://www.redbubble.com/people/nasa8x/works/18025880-python-programmer-t-shirt-and-hoodie?cat_context=u-tees&grid_pos=27&p=womens-fitted-scoop&rbs=af9a0cd0-2664-43de-bb85-e01a3e2009b4&ref=shop_grid

**Comment by honzajavorek** [comment]
Vypadá to, že než jsme vytvořili tričko české Python komunity, tak přestal existovat jack.cz. Chceme tohle oživit, nebo zavřít?

**Comment by jakolar** [comment]
@honzajavorek Ano, to je pravda, projekt jsme zavřeli. K zadání zaázky však nebyl potřeba a tiskárna OP Tiger stále funguje.
